### PR TITLE
[serve] Add code example for `get_deployment_handle` api

### PR DESCRIPTION
**Pull request by zcin**
## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by edoakes — COMMENTED** [review]
Can you add a second example showing how to get a non-ingress deployment?

**Review by edoakes — APPROVED** [review]

File: python/ray/serve/api.py [hunk 1]
```diff
@@ -801,6 +801,58 @@ def get_deployment_handle(
         RayServeException: If no Serve controller is running, or if
             calling from outside a Serve application and no application
             name is specified.
+
+    The following example gets the handle to the ingress deployment of
+    an application, which is equivalent to using `serve.get_app_handle`.
+
+    .. code-block:: python
```
Comment (GeneDer) [inline python/ray/serve/api.py:808]: I think in order for those code to be tested in doc test, we should add `>>>` in front of the code line instead. 
Re: [this doc](https://docs.ray.io/en/latest/ray-contribute/writing-code-snippets.html?highlight=code-block#doctest-style-examples) Tip `If you’re writing docstrings, exclude doctest:: to simplify your code.`
```
Example:
    >>> def is_even(x):
    ...     return (x % 2) == 0
    >>> is_even(0)
    True
    >>> is_even(1)
    False
```

Comment (zcin) [reply python/ray/serve/api.py]: Thanks @GeneDer! I switched to `testcode`, and confirmed that the code is being tested

**Review by GeneDer — APPROVED** [review]
:shipit:

**Comment by edoakes** [comment]
doctest failing

**Comment by zcin** [comment]
@edoakes Tests are passing

**Pull request merged**